### PR TITLE
Node Explorer: prompt to add nodes to SSH config file for VSCode remotes list

### LINKED ISSUE #217: Command to open Tailscale node in remote
While we can't add to the existing Remote SSH list, we can create a command that does something similar. Open a list of all Tailscale nodes that possibly have SSH enabled. Looks like we can separate these based on if they are reported to be online or not.

### PR DESCRIPTION
**Pull request by tendstofortytwo**
Fixes #217.

### CODE REVIEW

File: src/utils/sshconfig.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  const configStr = await vscode.workspace.fs
+    .readFile(sshConfigFilePath())
+    .then((a) => Buffer.from(a).toString('utf-8'));
+
+  return SSHConfig.parse(configStr);
+}
+
+export async function addToSSHConfig(configManager: ConfigManager, Host: string, User: string) {
+  const config = await readSSHConfig();
+  config.append({ Host, User, HostName: Host });
```
Comment (tylersmalley) [inline src/utils/sshconfig.ts:24]: We will need to update the config if the host is already present.

File: src/utils/sshconfig.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      const cfgHost = typeof h.Host === 'string' ? h.Host : h.Host[0];
+      if (cfgUsername !== tsUsername) {
+        const editHost = await vscode.window.showInformationMessage(
+          `The SSH config file specifies a username (${cfgUsername}) for host ${addr} that
+          is different from the SSH user configured in your Tailscale settings (${tsUsername}). Would you
+          like to update one of them?`,
+          'Update Tailscale settings',
+          'Update SSH config file',
+          'Do not update'
+        );
```
Comment (tylersmalley) [inline src/utils/sshconfig.ts:72]:
```suggestion
        const editHost = await vscode.window.showInformationMessage(
          `The SSH config file specifies a username (${cfgUsername}) for host ${addr} that
          is different from the SSH user configured in the Tailscale extension (${tsUsername}). Would you
          like to update one of them?`,
          'Update extension',
          'Update SSH config',
          'Do nothing'
        );
```

**Review by tylersmalley — CHANGES_REQUESTED** [review]
Looks great - just a few small suggestions. I think we could further improve the UX by removing the command and always syncing the SSH config when `tailscale.node.openRemoteCode` is run. It's the only thing that needs it, and in almost all situations it will be required.

File: src/config-manager.ts [hunk 3]
```diff
@@ -5,6 +5,8 @@ import * as path from 'path';
 interface Host {
   user: string;
   rootDir: string;
+  savedInSSHConfig?: boolean;
```
Comment (tylersmalley) [inline src/config-manager.ts:8]: I tend to prefer the intention here, something like `persistToSSHConfig`

File: src/config-manager.ts [hunk 4]
```diff
@@ -5,6 +5,8 @@ import * as path from 'path';
...
+  differentUserFromSSHConfig?: boolean;
```
Comment (tylersmalley) [inline src/config-manager.ts:9]: I tend to prefer the intention here, something like `persistToSSHConfig`

File: src/config-manager.ts (hunk 3 above)
Comment (tylersmalley) [inline src/config-manager.ts:8]: I tend to prefer the intention here, something like `persistToSSHConfig`

**Comment by tylersmalley** [comment]
One thing we have to figure out is how someone would change their selection if they choose to use a different SSH user. One thought is to have a context menu item that is only displayed if that setting is used - we could accomplish that with pattern matching on the context. So, in the event they choose to use the SSH config user, they would have a context button for "Update SSH config user" (wording could be improved)

**Comment by tendstofortytwo** [comment]
My thoughts there were that once the user decouples their Tailscale SSH user from the one in the SSH config file, the one in the config file is no longer our business - we use the user in the Tailscale settings to connect anyway.

File: src/node-explorer-provider.ts [hunk 5]
```diff
@@ -620,6 +622,10 @@ export class NodeExplorerProvider
       async (node: PeerRoot | FileExplorer) => {
         const { addr, path } = extractAddrAndPath(node);
 
+        if (addr && this.configManager.config.hosts?.[addr].persistToSSHConfig !== false) {
+          syncSSHConfig(addr, this.configManager);
```
Comment (tylersmalley) [inline src/node-explorer-provider.ts:626]: Lets await this

**Review by tylersmalley — APPROVED** [review]
Just one small change, then LGTM! This is a huge improvement for the user experience when using open in remote. Great work.

**Pull request merged**